libs/openFrameworks/app/ofAppGLFWWindow.cpp

```diff
@@ -322,6 +321,11 @@ shared_ptr<ofBaseRenderer> & ofAppGLFWWindow::renderer(){
 //--------------------------------------------
 void ofAppGLFWWindow::update(){
 	events().notifyUpdate();
+	
+	//show the window right before the first draw call.
+	if( settings.visible && windowP && ofGetFrameNum() == 0 ){
```

Comment:
can we better use the local frame number? that way it'll work to for windows opened after OF is running and doesn't use global calls which are not thread safe